idletiming_conn.go

```diff
@@ -50,7 +52,10 @@ func Conn(conn net.Conn, idleTimeout time.Duration, onIdle func()) *IdleTimingCo
 				atomic.StoreInt64(&c.lastActivityTime, time.Now().UnixNano())
 				continue
 			case <-timer.C:
-				onIdle()
+				c.Close()
```

Comment:
The only use case for idling a connection without closing it was in the unit test for idletiming.  That's silly, so now idletiming handles closing itself, but still provides an onIdle callback in case the consumer needs to do additional work like removing connections from a pool.